smpperf.go

```diff
@@ -175,8 +190,10 @@ func (s *SMPPerf) SendMessages() {
 			}
 			time.Sleep(r.Delay())
 			currTransceiver = (currTransceiver + 1) % s.NumSessions
+
+			// here we send the message
 			go func(transceiverIndex int) {
-				sm, err := transceivers[transceiverIndex].Submit(req)
+				sm, err := transceivers[transceiverIndex].transceiver.Submit(req)
```

Comment:
What about here, the class transceiver that you created can do the submission if it has a good connection. If it doesn't, the next transceiver can take the message.